DUI3-DX/Converters/ArcGIS/Speckle.Converters.ArcGIS3/Features/GisFeatureToSpeckleConverter.cs

```diff
@@ -2,41 +2,99 @@
 using Speckle.Core.Models;
 using Objects.GIS;
 using ArcGIS.Core.Data;
+using Speckle.Converters.ArcGIS3.Geometry;
+using ArcGIS.Core.Data.Exceptions;
 
 namespace Speckle.Converters.ArcGIS3.Features;
 
 public class GisFeatureToSpeckleConverter : IRawConversion<Row, GisFeature>
 {
   private readonly IRawConversion<ACG.Geometry, IReadOnlyList<Base>> _geometryConverter;
+  private readonly IGeometryUtils _geomUtils;
 
-  public GisFeatureToSpeckleConverter(IRawConversion<ACG.Geometry, IReadOnlyList<Base>> geometryConverter)
+  public GisFeatureToSpeckleConverter(
+    IRawConversion<ACG.Geometry, IReadOnlyList<Base>> geometryConverter,
+    IGeometryUtils geomUtils
+  )
   {
     _geometryConverter = geometryConverter;
+    _geomUtils = geomUtils;
   }
 
   public Base Convert(object target) => RawConvert((Row)target);
 
   public GisFeature RawConvert(Row target)
   {
-    var shape = (ACG.Geometry)target["Shape"];
-    var speckleShapes = _geometryConverter.RawConvert(shape).ToList();
-
     // get attributes
     var attributes = new Base();
     IReadOnlyList<Field> fields = target.GetFields();
-    int i = 0;
     foreach (Field field in fields)
     {
       string name = field.Name;
+      if (name != "Shape") // ignore the field with geometry itself
+      {
+        try
+        {
+          var value = target[name];
+          attributes[name] = value; // can be null
+        }
+        catch (ArgumentException)
+        {
+          // TODO: log in the conversion errors list
+          attributes[name] = null;
+        }
+      }
+    }
+
+    try
+    {
+      var shape = (ACG.Geometry)target["Shape"];
+      var speckleShapes = _geometryConverter.RawConvert(shape).ToList();
+
+      // re-shape the GisFeature:
+      // if shapes is a list of Meshes, set them as DisplayValue instead of geometry
```

Comment:
I think we should move away from these weird edge cases where the same function can return radically different things with different meaning.

Reply:
My problem here, is how do I assign displayValues to the GisFeature, if the best place to generate it, is within "Geometry converter" of GisFeature. And then, distribute returned geometry primitives into "geometry" (Point, Polyline, GisPolygon) and "displayValue" (Meshes) attributes. Maybe there is a better way?